pyiron_atomistics/atomistics/structure/neighbors.py

```diff
@@ -1081,3 +1083,101 @@ def get_volume_of_n_sphere_in_p_norm(n=3, p=2):
+
+    """
+    This class generates the neighbors for a given atomistic trajectory. The assumption here is that the trajectory is
+    canonical (no change in the number and types of species throughout the trajectory). The resulting indices,
+    distances, and vectors are stored as numpy arrays.
+
+    """
+
+    def __init__(self, has_structure, num_neighbors=12, table_name="neighbors_traj", **kwargs):
+        """
+
+        Args:
+            has_structure (:class:`.HasStructure`): object containing the structures to compute the neighbors on
+            num_neighbors (int): The cutoff for the number of neighbors
+            table_name (str): Table name for the base `DataContainer` (stores this object as a group in a
+                              HDF5 file with this name)
+            **kwargs (dict): Additional arguments to be passed to the `get_neighbors()` routine
+                             (eg. cutoff_radius, norm_order , etc.)
+        """
+        super().__init__(table_name=table_name)
+        self._has_structure = has_structure
+        self._neighbor_indices = None
+        self._neighbor_distances = None
+        self._neighbor_vectors = None
+        self._num_neighbors = num_neighbors
+        self._get_neighbors_kwargs = kwargs
+
+    @property
+    def neighbor_indices(self):
+        """
+        Neighbour indices (excluding itself) of each atom computed using the get_neighbors_traj() method
+
+        Returns:
+
+            numpy.ndarray/list: An array of dimension N_steps / stride x N_atoms x N_neighbors
+
+        """
+        return self._neighbor_indices
+
+    @property
+    def neighbor_distances(self):
+        """
+        Neighbour distances (excluding itself) of each atom computed using the get_neighbors_traj() method
+
+        Returns:
+
+            numpy.ndarray/list: An array of dimension N_steps / stride x N_atoms x N_neighbors
+
+        """
+        return self._neighbor_distances
+
+    @property
+    def neighbor_vectors(self):
+        """
+        Neighbour vectors (excluding itself) of each atom computed using the get_neighbors_traj() method
+
+        Returns:
+
+            numpy.ndarray/list: An array of dimension N_steps / stride x N_atoms x N_neighbors x 3
+
+        """
+        return self._neighbor_vectors
+
+    @property
+    def num_neighbors(self):
+        """
+        The maximum number of neighbors to be computed
+
+        Returns:
+
+            int: The max number of neighbors
+        """
+        return self._num_neighbors
+
+    def compute_neighbors(self):
+        """
+        Compute the neighbors across the trajectory
+        """
+        self._neighbor_indices, self._neighbor_distances, self._neighbor_vectors = \
+            _get_neighbors(self._has_structure,
+                           num_neighbors=self._num_neighbors, **self._get_neighbors_kwargs)
+
+
+def _get_neighbors(has_structure, num_neighbors=20, **kwargs):
+    n_steps = len(has_structure)
```

Comment:
```suggestion
    n_steps = has_structure.number_of_structures
```
like we discussed in the meeting.